### PR TITLE
create issue and pull request template.

### PR DESCRIPTION
**Pull request by BillWagner**
These will provide guidance when someone creates a new issue, or a new
pull request respectively.

/cc @mairaw @Rick-Anderson

### CODE REVIEW

File: ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+Issue Title
+===========
```
Comment (tdykstra) [inline ISSUE_TEMPLATE.md]: Why use this notation for headings instead of # for h1 ## for h2 etc.?

File: ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Your issue title is above. Please include the path to the
```
Comment (tdykstra) [inline ISSUE_TEMPLATE.md]: An example of including the path might be helpful.

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Insert long summary here.
+
+Explain your changes, and why you made them. If that
+information is already available in the issue referenced
+above, just referencing the issue is preferred to copying
+the text.
+
+This may not be necessary depending on the scope of the PR 
+changes. (For example, "fix typo in introduction.md" is
+sufficient to describe that PR).
```
Comment (tdykstra) [inline PULL_REQUEST_TEMPLATE.md]: put period inside closing paren

File: ISSUE_TEMPLATE.md (hunk 1 above)
Comment (BillWagner) [inline ISSUE_TEMPLATE.md]: I was going for simplicity, since I knew I wasn't going to have any headers deeper than 2. It was intentional, in that I wouldn't want to encourage issues or PR descriptions with more than 2 levels of headings. I believe we cover that we prefer the # notation in our contributing.md.

Thoughts from others?

File: ISSUE_TEMPLATE.md (hunk 2 above)
Comment (BillWagner) [inline ISSUE_TEMPLATE.md]: Updated. Interestingly, GitHub didn't match the update with your comment.

**Comment by BillWagner** [comment]
Removed the do-not-merge label. Please provide any feedback.

**Comment by tdykstra** [comment]
LGTM

File: ISSUE_TEMPLATE.md (hunk 1 above)
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: I'd go for consistency when possible.

File: ISSUE_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+General
+=======
+
+We use issues to drive the discussion for changes to
+existing topics, and the creation of new topics.
+
+This way, we have those discussions before significant
+effort is spent writing a new topic. Please see our
+Contributing.md guidelines for details.
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: link to the guide here?

File: ISSUE_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+topic with the issue. For example, if your issue is that
+our [list of samples](https://stage.docs.microsoft.com/en-us/dotnet/articles/samples-and-tutorials/index)
+is missing a sample, you'd put [samples-and-tutorials/index.md]
+in the issue title.
+
+If your issue is that a topic is missing, please preface
+the issue title with [NewTopic]. That helps us find requests
+for new topics.
+
+General
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: should this come first even before the title as a header like Issue Template? And then you could to the opening paragraph, this article talks about some guidelines about opening issues in our repo, etc.

File: ISSUE_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+===========
+
+Your issue title is above. Please include the path to the
+topic with the issue. For example, if your issue is that
+our [list of samples](https://stage.docs.microsoft.com/en-us/dotnet/articles/samples-and-tutorials/index)
+is missing a sample, you'd put [samples-and-tutorials/index.md]
+in the issue title.
+
+If your issue is that a topic is missing, please preface
+the issue title with [NewTopic]. That helps us find requests
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: aren't we going to use the label for that? isn't that easier to sort than in the issue title?

File: ISSUE_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+Contributing.md guidelines for details.
+
+Issues with Existing Topics
+===========================
+
+Please describe the problem with the current topic. Be as
+specific as you can. 
+
+- Is content that you wanted missing?
+- Is content incorrect?
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: incorrect or outdated?

File: ISSUE_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+- Are links broken?
+- Is some content not relevant to the topic?
+
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: Maybe you could add: Are we missing a scenario for a particular topic? Do we need to add samples or images to a topic?

File: ISSUE_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+This way, we have those discussions before significant
+effort is spent writing a new topic. Please see our
+Contributing.md guidelines for details.
+
+Issues with Existing Topics
+===========================
+
+Please describe the problem with the current topic. Be as
+specific as you can.
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: Maybe add: Add a link to the topic?

File: ISSUE_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
+Issue Title
+===========
+
+Your issue title is above. Please include the path to the
+topic with the issue. For example, if your issue is that
+our [list of samples](https://stage.docs.microsoft.com/en-us/dotnet/articles/samples-and-tutorials/index)
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: link should go to production

File: ISSUE_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+is missing a sample, you'd put [samples-and-tutorials/index.md]
```
Comment (mairaw) [inline ISSUE_TEMPLATE.md]: should we be doing this instead? we're kind of asking them to know our structure. I prefer they'd add a link in the description and we can change the title later if we want.

File: ISSUE_TEMPLATE.md (hunk 6 above)
Comment (Rick-Anderson) [inline ISSUE_TEMPLATE.md]: I'm not sure we need to have customers set the title to path/doc.md, and even if we ask them I'm guessing few will follow this. But I think it's critical that issues are stored that way for easy searching and to prevent duplicates. They is frequently an issue open and you want to add a task list
- [ ] Fix xyz
- [ ]  add section on ABC

It doesn't hurt to ask folks to use this format. Not organizing topic issues this way makes it difficult to manage issues.

File: PULL_REQUEST_TEMPLATE.md [hunk 12]
```diff
@@ -0,0 +1,32 @@
+Fixes #Issue_Number
```
Comment (mairaw) [inline PULL_REQUEST_TEMPLATE.md]: Can we add a title here? Also this explanation about using the Fixes notation should be moved to details no? Maybe link to GitHub help for more info on this: https://help.github.com/articles/closing-issues-via-commit-messages/

File: PULL_REQUEST_TEMPLATE.md [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+>note: The "Fixes #nnn" syntax in the PR description causes
+>GitHub to close the issue when this PR is merged. It saves
+>all of us some work. Not all PRs need to reference an issue.
+>See CONTRIBUTING.md for details.
```
Comment (mairaw) [inline PULL_REQUEST_TEMPLATE.md]: link to contributing guide?

File: PULL_REQUEST_TEMPLATE.md [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+Summary
+========
+
+Insert short (less than 140 character) summary here.
+
+Please always include this.
```
Comment (mairaw) [inline PULL_REQUEST_TEMPLATE.md]: I'm confused on what you mean about insert short summary here and long summary here.

**Comment by Rick-Anderson** [comment]
tldr;
If it's too long, few will read it. Lets put the critical info at the top. Folks wanting to create a new topic will read the details section - but most folks creating an issue (on a problem) probably won't read anything.

**Comment by mairaw** [comment]
I wasn't familiar with these GitHub templates. Thanks Bill for clarifying that offline. My suggestion then based on what I read [here](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/), would be to move these files to the hidden folder. And maybe we should also move some of the other templates/style guides to a different folder, so our root looks a bit cleaner.

File: ISSUE_TEMPLATE.md (hunk 5 above)
Comment (BillWagner) [inline ISSUE_TEMPLATE.md]: I'd prefer keeping the title guidance at the very top. That way it's as close to where a reader would type the new title as possible.

I'm working on making this template much shorter (see @Rick-Anderson 's comments below). I think that will address this comment as well.

File: ISSUE_TEMPLATE.md (hunk 6 above)
Comment (BillWagner) [inline ISSUE_TEMPLATE.md]: I tried to simplify the recommendations here. I'm thinking that if it's close, we can normalize it.

File: ISSUE_TEMPLATE.md (hunk 11 above)
Comment (BillWagner) [inline ISSUE_TEMPLATE.md]: I think you're right. If they include a link in the description, we can edit the title.

**Comment by BillWagner** [comment]
I tried to respond to all feedback, and make the issue template as short as possible. I think another review is in order. @mairaw @Rick-Anderson

File: .github/ISSUE_TEMPLATE.md [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+Be as descriptive as you can with your title. If your issue is requesting
+a new topic, please preface the title with [NewTopic].
+
+# General
+
+We use issues to drive the discussion for changes to
+existing topics, and the creation of new topics.
+
+This way, we have those discussions before significant
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: Shouldn't new topics be a different section? What's the purpose of this doc? For all issues? Don't you want a new topic section? It would be good to have links to each section at the top. This text should be written as:

something like:

An outline and discussion should take place before significant effort on a topic.

File: .github/ISSUE_TEMPLATE.md [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+effort is spent writing a new topic. Please see our
+[Contributing Guide](https://github.com/dotnet/core-docs/blob/master/CONTRIBUTING.md)
+for details.
+
+## Issues with Existing Topics
+
+Please include a link to the topic. Describe the problem as specifically as you can. Consider:
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: Do you want to mention they don't need to create an issue for a typo or simple correction. Just submit a PR with the fix.

File: .github/ISSUE_TEMPLATE.md [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+where you looked for this information before opening an
+issue.
+
+1. Write an abstract
+
+   - In one **short** paragraph, describe what this topic will cover.
+
+2. Fill in an outline
+
+    - Fill in as much as you can about how someone should cover the topic. We'll help review the outline and approve that before anyone writes a  topic.
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: as much as you can is nebulous.
Provide A complete outline.

File: .github/ISSUE_TEMPLATE.md [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+
+3. Suggest reviewers
+
+    - If you know who should be looking at this topic, please '@' mention them here. If you're not sure, just leave this blank. The core content team looks at new issues regularly.
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: If you know someone who can provide feedback, use @<alias> to ask them to review.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 19]
```diff
@@ -0,0 +1,37 @@
+# Title
+
+Please write a descriptive title that describes
```
Comment (Rick-Anderson) [inline .github/PULL_REQUEST_TEMPLATE.md]: ~~Please~~ - pretty much everywhere. it's not needed.

Comment (Rick-Anderson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Provide a title that describes ...

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+
+Fixes #Issue_Number
+
+>note: The "Fixes #nnn" syntax in the PR description causes
+>GitHub to close the issue when this PR is merged. See 
+> [this GitHub help document](https://help.github.com/articles/closing-issues-via-commit-messages/)
+> for details. It saves all of us some work. Not all PRs need to reference an issue.
+>See [our contributing guide](https://github.com/dotnet/core-docs/blob/master/CONTRIBUTING.md) for details.
+
+Insert short (less than 140 character) summary here.
```
Comment (Rick-Anderson) [inline .github/PULL_REQUEST_TEMPLATE.md]: It'd not twitter. One or two sentences.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+
+Please always include this.
```
Comment (Rick-Anderson) [inline .github/PULL_REQUEST_TEMPLATE.md]: delete

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+Explain your changes, and why you made them. If that
+information is already available in the issue referenced
+above, just referencing the issue is preferred to copying
+the text.
+
+This may not be necessary depending on the scope of the PR 
+changes. (For example, "fix typo in introduction.md" is
+sufficient to describe that PR.)
+
+# Suggested Reviewers
```
Comment (Rick-Anderson) [inline .github/PULL_REQUEST_TEMPLATE.md]: Remove first mention or remove this. Most folks won't have reviewers.

File: .github/ISSUE_TEMPLATE.md [hunk 23]
```diff
@@ -0,0 +1,41 @@
+# Issue Title
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: Consider two template or better splitting this up between issue and PR. Internal links at the top work.

File: .github/ISSUE_TEMPLATE.md [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+
+This way, we have those discussions before significant
+effort is spent writing a new topic. Please see our
+[Contributing Guide](https://github.com/dotnet/core-docs/blob/master/CONTRIBUTING.md)
+for details.
+
+## Issues with Existing Topics
+
+Please include a link to the topic. Describe the problem as specifically as you can. Consider:  
+
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: I'm not sure we need to enumerate a few possible reasons. People find and issue then decide to report it. What's the benefit of providing this list?

File: .github/ISSUE_TEMPLATE.md [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+- Is content incorrect or outdated?
```
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: perhaps, what is wrong or how could we improve ...

File: .github/ISSUE_TEMPLATE.md (hunk 23 above)
Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: We do have one template for each @Rick-Anderson.

File: .github/ISSUE_TEMPLATE.md (hunk 15 above)
Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: This text will show up inside the Write box when you're creating an issue @Rick-Anderson. Similar to what docfx GH has: 
https://github.com/dotnet/docfx/issues/new

Adding links from the top just adds more text to that box. We should keep it to a minimal.

File: .github/ISSUE_TEMPLATE.md (hunk 16 above)
Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: I think it'd be good.

File: .github/ISSUE_TEMPLATE.md (hunk 24 above)
Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: Considering where this shows up, I'd remove the list too.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+Please write a descriptive title that describes
+what you've fixed (or created) with this Pull Request.
+
+# Summary
+
+Fixes #Issue_Number
+
+>note: The "Fixes #nnn" syntax in the PR description causes
+>GitHub to close the issue when this PR is merged. See 
+> [this GitHub help document](https://help.github.com/articles/closing-issues-via-commit-messages/)
```
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: since this will show up as-is in that box, I'd remove links. Sorry for asking you that yesterday without understanding how this works. :disappointed:

File: .github/ISSUE_TEMPLATE.md (hunk 15 above)
Comment (Rick-Anderson) [inline .github/ISSUE_TEMPLATE.md]: That looks great. Why not copy it?

Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: That looks more like a product issue template. I think ours should be a bit different.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 27]
```diff
@@ -0,0 +1,37 @@
+# Title
+
+Please write a descriptive title that describes
+what you've fixed (or created) with this Pull Request.
```
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: Add PR in parentheses here so you can refer as just PR later

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 26 above)
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: My suggestion for this note:
```
> Note: The "Fixes #nnn" syntax in the PR description causes
> GitHub to automatically close the referenced issue when this PR is merged. 
> Remove that line if you don't have issues associated with this PR.
> Click on the guidelines for contributing link above for details.
```
Something along these lines.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 20 above)
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: Do we have to say here? I'd also move this up before the fixes notation.
Maybe just say: Provide a short description of what you've done.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 28]
```diff
@@ -0,0 +1,37 @@
...
+> for details. It saves all of us some work. Not all PRs need to reference an issue.
+>See [our contributing guide](https://github.com/dotnet/core-docs/blob/master/CONTRIBUTING.md) for details.
+
+Insert short (less than 140 character) summary here.
+
+Please always include this.
+
+## Details
+
+Insert long summary here.
```
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'd remove this too.

**Comment by BillWagner** [comment]
One more draft up there. I think they are both getting close.

File: .github/ISSUE_TEMPLATE.md [hunk 29]
```diff
@@ -0,0 +1,36 @@
...
+We use issues to drive the discussion for changes to
+existing topics and the creation of new topics.
+
+This way, we have those discussions before significant
+effort is spent writing a new topic. Click on the Guidelines
+for Contributing link above for details.
+
+## Issues with Existing Topics
+
+If the issue is a simple typo or similar correction, just submit it PR.
```
Comment (tdykstra) [inline .github/ISSUE_TEMPLATE.md]: just submit it PR --> just submit a PR

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 30]
```diff
@@ -0,0 +1,30 @@
+# Title
+
+Provide title that describes
```
Comment (tdykstra) [inline .github/PULL_REQUEST_TEMPLATE.md]: provide title that describes --> provide a title that describes

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 31]
```diff
@@ -0,0 +1,30 @@
...
+what you've fixed (or created) with this Pull Request (PR).
+
+## Summary
+
+Fixes #Issue_Number
+
+>note: The "Fixes #nnn" syntax in the PR description causes
```
Comment (tdykstra) [inline .github/PULL_REQUEST_TEMPLATE.md]: is "note:" deliberately lower-case?

Comment (BillWagner) [inline .github/PULL_REQUEST_TEMPLATE.md]: Yes, but that doesn't mean it's correct. :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 32]
```diff
@@ -0,0 +1,30 @@
...
+>GitHub to automatically close the issue when this PR is merged.
+> Remove that line if you don't have issues associated with this
+> PR. Click on the Guidelines for Contributing link above for details.
+
+Insert short (one or two sentence) summary here.
```
Comment (tdykstra) [inline .github/PULL_REQUEST_TEMPLATE.md]: Insert short --> Insert a short

File: .github/ISSUE_TEMPLATE.md [hunk 33]
```diff
@@ -0,0 +1,36 @@
...
+We use issues to drive the discussion for changes to
+existing topics and the creation of new topics.
+
+This way, we have those discussions before significant
+effort is spent writing a new topic. Click on the Guidelines
+for Contributing link above for details.
+
+## Issues with Existing Topics
+
+If the issue is a simple typo or similar correction, just submit a PR.
```
Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: Instead of "just submit a PR", maybe we should say you can just submit a PR. They can still open an issue if they prefer IMO,

File: .github/ISSUE_TEMPLATE.md [hunk 34]
```diff
@@ -0,0 +1,36 @@
...
+Otherwise, let us know what's wrong or what we should improve. Include a
+link to the topic.
+
+## Requests for new Topics
+
+Tell us where this topic should go in the Table of Contents. Consider where you
```
Comment (mairaw) [inline .github/ISSUE_TEMPLATE.md]: should this be one of the steps? I don't think it should be the first thing when requesting a new topic.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 35]
```diff
@@ -0,0 +1,30 @@
...
+## Summary
+
+Fixes #Issue_Number
+
+>Note: The "Fixes #nnn" syntax in the PR description causes
+>GitHub to automatically close the issue when this PR is merged.
+> Remove that line if you don't have issues associated with this
+> PR. Click on the Guidelines for Contributing link above for details.
+
+Insert a short (one or two sentence) summary here.
```
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'd move this up before the Fixes.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 36]
```diff
@@ -0,0 +1,30 @@
+# Title
+
+Provide a title that describes
```
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: Do we need this? Since the title is mandatory they won't miss it. Or maybe just say:
"On the Title above, describe what you've ....

**Comment by mairaw** [comment]
Looks much better! Left a couple of minor comments.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 37]
```diff
@@ -0,0 +1,30 @@
...
+
+## Details
+
+Explain your changes, and why you made them. If that
+information is already available in the issue referenced
+above, just referencing the issue is preferred to copying
+the text.
+
+This may not be necessary depending on the scope of the PR 
+changes. (For example, "fix typo in introduction.md" is
```
Comment (mairaw) [inline .github/PULL_REQUEST_TEMPLATE.md]: don't think you need the parentheses here

**Comment by mairaw** [comment]
:ship: :it: :fireworks:

**Comment by BillWagner** [comment]
Merging and shipping. :fireworks:
